File: AlgoExpert/python/tandem_bicycle.py
"""
Difficulty: Easy
"""


def tandemBicycle(redShirtSpeeds, blueShirtSpeeds, fastest):
    # Write your code here.
    redShirtSpeeds.sort()
    if fastest:
        blueShirtSpeeds.sort(reverse=True)
    else:
        blueShirtSpeeds.sort()

    total_speed = 0
    for red, blue in zip(redShirtSpeeds, blueShirtSpeeds):
        total_speed += max(red, blue)

    return total_speed

File: AlgoExpert/cpp/tandem_bicycle.cc
/* Notes
Difficulty: Easy
*/

#include <algorithm>
#include <vector>

int tandemBicycle(std::vector<int> redShirtSpeeds,
                  std::vector<int> blueShirtSpeeds, bool fastest) {
  // Write your code here.
  std::sort(redShirtSpeeds.begin(), redShirtSpeeds.end());
  if (fastest) {
    std::sort(blueShirtSpeeds.rbegin(), blueShirtSpeeds.rend());
  } else {
    std::sort(blueShirtSpeeds.begin(), blueShirtSpeeds.end());
  }

  int total_speed = 0;
  for (size_t idx = 0; idx < redShirtSpeeds.size(); ++idx) {
    total_speed += std::max(redShirtSpeeds[idx], blueShirtSpeeds[idx]);
  }

  return total_speed;
}